.github/scripts/get_merge_base_info.py

```diff
@@ -0,0 +1,70 @@
+from typing import Dict, Any, List
+import subprocess
+from pathlib import Path
+from utils import query_rockset, upload_to_rockset
+
+REPO_ROOT = Path(__file__).resolve().parent.parent.parent
+
+FAILED_TEST_SHAS_QUERY = """
+SELECT
+    DISTINCT j.head_sha,
+FROM
+    commons.failed_tests_run t
+    join workflow_job j on t.job_id = j.id
+    left outer join commons.merge_bases mb on j.head_sha = mb.sha
+where
+    mb.merge_base is null
```

Comment:
Wow, this query runs way faster than I expected!

Reply:
The failed_tests_run and merge_bases are small.  I assume in the future we'll need to gate it on time, and figure out some way to balance old ratings with new ones, but for now it seems ok.

Also putting this here because I don't want to make another comment: added some very simple tests, but I think updating them might get annoying/end up with just assuming the output is correct if we change the rating system a lot.